system/lib/wasmfs/js_api.cpp

```diff
@@ -126,25 +126,25 @@ int _wasmfs_mkdir(char* path, int mode) {
 int _wasmfs_rmdir(char* path){ return __syscall_unlinkat(AT_FDCWD, (intptr_t)path, AT_REMOVEDIR); }
 
 int _wasmfs_open(char* path, int flags, mode_t mode) {
-  int err = __syscall_openat(AT_FDCWD, (intptr_t)path, flags, mode);
-  if (err == -1) {
-    return -errno;
-  }
-  return err;
+  return __syscall_openat(AT_FDCWD, (intptr_t)path, flags, mode);
 }
 
 int _wasmfs_mknod(char* path, mode_t mode, dev_t dev) {
-  int err = __syscall_mknodat(AT_FDCWD, (intptr_t)path, mode, dev);
-  if (err == -1) {
-    return errno;
-  }
-  return err;
+  return __syscall_mknodat(AT_FDCWD, (intptr_t)path, mode, dev);
 }
 
 int _wasmfs_unlink(char* path) {
   return __syscall_unlinkat(AT_FDCWD, (intptr_t)path, 0);
 }
 
+int _wasmfs_truncate(char *path, long size) {
+  return __syscall_truncate64((intptr_t)path, size);
+}
+
+int _wasmfs_ftruncate(int fd, long size) {
+  return __syscall_ftruncate64(fd, size);
+}
+
```

Comment:
Since there's no extra logic necessary here, is it possible to skip these wrappers and just call the underlying syscall directly from the JS?

Reply:
That makes sense, although I'm having trouble figuring out how to import the syscalls to `library_wasmfs.js`. I've tried using `__syscall_truncate64` and `___syscall_truncate64` (Three underscores) and neither worked. Normally, I would add `_wasmfs_truncate` to `emcc.py`, but I don't think it worked. How should I go about making the syscall available to call?

Reply:
Doing this as a follow-up sounds good to me.